web/src/lib/roadmap/roadmapBuilder.ts

```diff
@@ -128,37 +131,32 @@ const findTaskInRoadmapByFilename = (
   roadmapBuilder: RoadmapBuilder,
   taskFilename: string
 ): TaskBuilder | undefined => {
-  for (const step of roadmapBuilder.steps) {
-    const found = step.tasks.find((task) => task.filename === taskFilename);
-    if (found) return found;
-  }
+  const found = roadmapBuilder.tasks.find((task) => task.filename === taskFilename);
+  if (found) return found;
 };
 
 const convertToRoadmap = async (roadmapBuilder: RoadmapBuilder): Promise<Roadmap> => {
   const roadmap = {
-    steps: await Promise.all(
-      roadmapBuilder.steps.map(async (step) => ({
-        ...step,
-        tasks: await Promise.all(
-          step.tasks.sort(orderByWeight).map((task) => fetchTaskByFilename(task.filename))
-        ),
-      }))
+    steps: await Promise.all(roadmapBuilder.steps),
```

Comment:
this doesn't need to be an `await Promise.all` anymore, for steps here, because it's no long an asynchronous function